addons/sale/views/product_views.xml

```diff
@@ -122,4 +122,17 @@
         </field>
     </record>
 
+    <record id="product_view_search_catalog" model="ir.ui.view">
```

Comment:
It could also work for purchase orders but I think it's not really useful in this case. So for now, I'll follow your suggestion and make it invisible when we're not updating the sale order line :+1: